tracing/tracers/otel/otel.go

```diff
@@ -0,0 +1,299 @@
+package otel
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/lightstep/lightstep-tracer-go"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	otelBridge "go.opentelemetry.io/otel/bridge/opentracing"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracehttp"
+	"go.opentelemetry.io/otel/propagation"
+	"go.opentelemetry.io/otel/sdk/resource"
+	sdk "go.opentelemetry.io/otel/sdk/trace"
+	semconv "go.opentelemetry.io/otel/semconv/v1.4.0"
+	"go.opentelemetry.io/otel/trace"
+
+	ot "github.com/opentracing/opentracing-go"
+)
+
+const (
+	defServiceName = "skipper"
+)
+
+type tracerBuilder struct {
+	exporter *otlptrace.Exporter
+	provider *sdk.TracerProvider
+	opt      options
+}
+
+type options struct {
+	customHeaders  map[string]string
+	host           string
+	port           int
+	useHTTP        bool
+	ServiceName    string
+	ServiceVersion string
+	Environment    string
+	customPath     string
+	insecure       bool
+
+	exportTimeout      time.Duration
+	batchTimeout       time.Duration
+	maxExportBatchSize int
+	maxSpanQueueSize   int
+	compressionMethod  string
+	tags               map[string]string
+}
+
+func WithBridgeTracer(ctx context.Context, opts []string) (trace.Tracer, ot.Tracer, error) {
+	opt, err := parseOptions(opts)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	builder := tracerBuilder{
+		opt: opt,
+	}
+
+	t, err := builder.Build(ctx)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	_, otTracer, wrapper := otelBridge.NewTracerPairWithContext(ctx, t)
+	otel.SetTracerProvider(wrapper)
+	return t, otTracer, nil
+}
+
+func parseOptions(opts []string) (options, error) {
+	config := options{}
+	globalTags := make(map[string]string)
+
+	for _, o := range opts {
+		key, val, _ := strings.Cut(o, "=")
+		switch key {
```

Comment:
Should we return and error when `key == "someDefinitelyWrongKey"`?

Reply:
I also agree that this behavior is desirable, but all other tracers just ignore keys that don't exist and fails only when a required value is not provided, so I just followed the same behavior for otel tracer.